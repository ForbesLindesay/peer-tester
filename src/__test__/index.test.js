import runTestScript from '../';

test('logAllResults: true', async () => {
  let output = '';
  const result = await runTestScript(__dirname + '/../../example', {
    log: message => output += message + '\n',
    logAllResults: true,
  });
  expect(output).toMatchSnapshot();
  expect(result).toBe(1);
});

test('logAllResults: false', async () => {
  let output = '';
  const result = await runTestScript(__dirname + '/../../example', {
    log: message => output += message + '\n',
  });
  expect(output).toMatchSnapshot();
  expect(result).toBe(1);
});

test('logAllResults: false, color: true', async () => {
  let output = '';
  const result = await runTestScript(__dirname + '/../../example', {
    log: message => output += message + '\n',
    color: true,
  });
  expect(output).toMatchSnapshot();
  expect(result).toBe(1);
});
