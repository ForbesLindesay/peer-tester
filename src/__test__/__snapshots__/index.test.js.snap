// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`logAllResults: false 1`] = `
"Installing promise@6.0.0
Installing promise@6.0.1
Installing promise@6.1.0
Testing promise@6.0.0
Testing promise@6.0.1
Testing promise@6.1.0

Failed promise@6.0.1
> example@ test <cwd>/.peer-tester/promise_v6.0.1
> node test

promise version v6.0.1
Simulating faield test for v6.0.1
npm ERR! Test failed.  See above for more details.
"
`;

exports[`logAllResults: false, color: true 1`] = `
"Installing promise@6.0.0
Installing promise@6.0.1
Installing promise@6.1.0
Testing promise@6.0.0
Testing promise@6.0.1
Testing promise@6.1.0
[31m
Failed promise@6.0.1[39m
> example@ test <cwd>/.peer-tester/promise_v6.0.1
> node test

promise version v6.0.1
Simulating faield test for v6.0.1
npm ERR! Test failed.  See above for more details.
"
`;

exports[`logAllResults: true 1`] = `
"Installing promise@6.0.0
Installing promise@6.0.1
Installing promise@6.1.0
Testing promise@6.0.0
Testing promise@6.0.1
Testing promise@6.1.0

Succeeded promise@6.0.0
> example@ test <cwd>/.peer-tester/promise_v6.0.0
> node test

promise version v6.0.0

Failed promise@6.0.1
> example@ test <cwd>/.peer-tester/promise_v6.0.1
> node test

promise version v6.0.1
Simulating faield test for v6.0.1
npm ERR! Test failed.  See above for more details.

Succeeded promise@6.1.0
> example@ test <cwd>/.peer-tester/promise_v6.1.0
> node test

promise version v6.1.0
"
`;
